vllm/model_executor/parallel_utils/custom_all_reduce.py

```diff
@@ -134,18 +141,48 @@ def _is_full_nvlink(rank, world_size):
 
 
 def _can_p2p(rank: int, world_size: int) -> bool:
+    num_dev = torch.cuda.device_count()
+    # note: num dev can be larger than world_size if we're only using
+    # first few GPUs
```

Comment:
This is the case when more than one nodes (hosts) are used for TP, right?

Reply:
No. This check is for the case that the users have N GPUs, but are only using the first M GPUs where M < N.